### PR TITLE
PERF: more flexible iso8601 parsing

### LINKED ISSUE #9714: to_datetime 1000x slower for timezone-aware strings vs timezone-agnostic
When converting a string date column to datetime, if the string has a GMT timezone suffix (e.g. "-0800"), it takes 1000x longer to parse:

    dates = pd.Series(pd.date_range('1/1/2000', periods=2000))
    string_dates = dates.apply(lambda s: str(s))
    tz_string_dates = string_dates.apply(lambda dt: dt + ' -0800')

    %timeit pd.to_datetime(string_dates)
    > 1000 loops, best of 3: 579 µs per loop
    %timeit pd.to_datetime(tz_string_dates)
    > 1 loops, best of 3: 562 ms per loop

Note microseconds vs milliseconds. 3 orders of magnitude... seems unnecessary. This can make loading CSVs into correctly-typed dataframes very, very, very slow for large datasets.

### LINKED ISSUE #11871: BUG: to_datetime issue parsing non-zero padded month in 0.17.1
In pandas 0.16.2, the following date (non-zero padded month) was parsing correctly:

```
>>> import pandas
>>> pandas.__version__
'0.16.2'
>>> pandas.to_datetime('2005-1-13', format='%Y-%m-%d')
Timestamp('2005-01-13 00:00:00')
```

With 0.17.1, it raises a ValueError:

```
>>> import pandas
>>> pandas.__version__
u'0.17.1'
>>> pandas.to_datetime('2005-1-13', format='%Y-%m-%d')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/dpinte/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/pandas/util/decorators.py", line 89, in wrapper
    return func(*args, **kwargs)
  File "/Users/dpinte/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/pandas/tseries/tools.py", line 276, in to_datetime
    unit=unit, infer_datetime_format=infer_datetime_format)
  File "/Users/dpinte/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/pandas/tseries/tools.py", line 397, in _to_datetime
    return _convert_listlike(np.array([ arg ]), box, format)[0]
  File "/Users/dpinte/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/pandas/tseries/tools.py", line 383, in _convert_listlike
    raise e
ValueError: time data '2005-1-13' does match format specified
```

Even if `%m` is supposed to be used for zero-padded month definitions, Python's strptime function parses them properly.

Is this a known issue?

### LINKED ISSUE #11899: PERF: allow even more flexible ISO 8601 datetime parsing
Hello,

I noticed that there is a huge code speed difference with `to_datetime` execution when format is not given and when it's given.

I wonder if there is not some room for improvements here!

```
In [1]: %time df=pd.read_csv("AUDUSD-2014-01.csv", names=['Symbol', 'Date', 'Bid', 'Ask'])
CPU times: user 3.31 s, sys: 481 ms, total: 3.79 s
Wall time: 4.13 s

In [2]: df
Out[274]:
          Symbol                   Date      Bid      Ask
0        AUD/USD  20140101 21:55:34.404  0.88796  0.88922
1        AUD/USD  20140101 21:55:34.444  0.88805  0.88914
2        AUD/USD  20140101 21:55:34.475  0.88809  0.88910
3        AUD/USD  20140101 21:55:48.962  0.88811  0.88908
4        AUD/USD  20140101 21:56:38.293  0.88808  0.88887
...          ...                    ...      ...      ...
1947101  AUD/USD  20140131 21:59:48.048  0.87525  0.87589
1947102  AUD/USD  20140131 21:59:54.599  0.87527  0.87589
1947103  AUD/USD  20140131 21:59:56.927  0.87531  0.87588
1947104  AUD/USD  20140131 21:59:59.365  0.87531  0.87574
1947105  AUD/USD  20140131 22:00:00.038  0.87531  0.87574

[1947106 rows x 4 columns]

In [3]: %time pd.to_datetime(df['Date'])
CPU times: user 11min 44s, sys: 19.4 s, total: 12min 4s
Wall time: 12min 46s
Out[3]:
0         2014-01-01 21:55:34.404
1         2014-01-01 21:55:34.444
2         2014-01-01 21:55:34.475
3         2014-01-01 21:55:48.962
4         2014-01-01 21:56:38.293
                    ...
1947101   2014-01-31 21:59:48.048
1947102   2014-01-31 21:59:54.599
1947103   2014-01-31 21:59:56.927
1947104   2014-01-31 21:59:59.365
1947105   2014-01-31 22:00:00.038
Name: Date, dtype: datetime64[ns]

In [4]: fmt='%Y%m%d %H:%M:%S.%f'

In [5]: %time pd.to_datetime(df['Date'], format=fmt)
CPU times: user 37.3 s, sys: 1.31 s, total: 38.6 s
Wall time: 40 s

In [6]: timedelta(minutes=12, seconds=46) / timedelta(seconds=40)
Out[6]: 19.15
```

There is x19.15 factor!!!

Sample data can be found here
https://drive.google.com/file/d/0B8iUtWjZOTqla3ZZTC1FS0pkZXc/view?usp=sharing

See also https://github.com/pydata/pandas-datareader/issues/153

### PR DESCRIPTION
**Pull request by chris-b1**
closes #9714
closes #11899
closes #11871

makes ISO parser in C handle the following. I think 2 & 3 aren't actually iso8601 anymore, but close and unambiguous.
1. dates without '-' separator 
2. dates without space before tz
3. dates without leading 0s in month/day
   (this ONLY works if the date has separators, eg. `"2015-1-1"` parses, but `"201511"` doesn't because it's ambiguous)

asv results - adds a small amount of overhead to the standard case (`'2014-01-01'`)

```
    before     after       ratio
  [1ae6384 ] [e922b05 ]
     5.17ms     5.34ms      1.03  timeseries.timeseries_to_datetime_iso8601.time_timeseries_to_datetime_iso8601
     5.15ms     5.50ms      1.07  timeseries.timeseries_to_datetime_iso8601.time_timeseries_to_datetime_iso8601_format
-  111.89ms     5.27ms      0.05  timeseries.timeseries_to_datetime_iso8601.time_timeseries_to_datetime_iso8601_format_no_sep
-     2.02s     5.33ms      0.00  timeseries.timeseries_to_datetime_iso8601.time_timeseries_to_datetime_iso8601_nosep
-     2.97s   218.25ms      0.07  timeseries.timeseries_to_datetime_iso8601.time_timeseries_to_datetime_iso8601_tz_spaceformat

```

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
A question: is is worth the added complexity to the ISO 8601 C-parser to be able to handle the dates without separator (which is actually not really ISO 8601 anymore), if when passing the `format` is already as fast to parse this? 
Of course, you then need to specify this, but this could also be alleviated by making `infer_datetime_format=True` the default.

File: asv_bench/benchmarks/timeseries.py [hunk 1]
```diff
@@ -1059,33 +1059,26 @@ class timeseries_to_datetime_iso8601(object):
...
-        self.N = 100000
-        self.rng = date_range(start='1/1/2000', periods=self.N, freq='T')
-        if hasattr(Series, 'convert'):
-            Series.resample = Series.convert
-        self.ts = Series(np.random.randn(self.N), index=self.rng)
         self.rng = date_range(start='1/1/2000', periods=20000, freq='H')
-        self.strings = [x.strftime('%Y-%m-%d %H:%M:%S') for x in self.rng]
+        self.strings = self.rng.strftime('%Y-%m-%d %H:%M:%S')
+        self.strings_nosep = self.rng.strftime('%Y%m%d %H:%M:%S')
+        self.strings_tz_space = pd.Series(self.rng.strftime('%Y-%m-%d %H:%M:%S')) + ' -0800'
```
Comment (jorisvandenbossche) [inline asv_bench/benchmarks/timeseries.py]: The `strftime` method on a DatetimeIndex is rather new. I don't really know what our policy on this should be, but I think this will make it difficult to run the asv benchmarks for a longer time back?

**Comment by chris-b1** [comment]
Hard to say if it's worth it (re dates with separators), not sure it's a super common format.  But there is a substantial speedup even over providing a format to the strptime parser (20x in that asv bench).  

Not that it really matters, but from I've read dates without separators would be considered ISO 8601, e.g. http://www.cl.cam.ac.uk/~mgk25/iso-time.html.

**Comment by jreback** [comment]
how hard would it to allow a selection of separators

eg -, ., /, \ (assume that if one is found it would have to repeat for the date portions, iow don't allow mixed seps)

w/o sacrificing ambiguity?

**Comment by jreback** [comment]
also, let's add another issue to make ``infer_datetime_format=True`` as the default for ``to_datetime`` and ``read_csv``. I don't think this will impact anything and just make more things work.

**Comment by jorisvandenbossche** [comment]
@chris-b1 Ah, yes, I missed that you also passed this to the ISO parser if that format is given. I misinterpreted the asv output so thinking that the generic format parser was as fast as the ISO parser, but indeed quite a bit slower.

**Comment by chris-b1** [comment]
@jreback - I don't think it would be too bad to parse multiple separators, I'll try it out

File: asv_bench/benchmarks/timeseries.py (hunk 1 above)
Comment (jreback) [inline asv_bench/benchmarks/timeseries.py]: i agree here. why don't you leave the generation of the test cases using the list comprehension (how ``self.strings`` was)

**Comment by jreback** [comment]
@chris-b1 just some minor comments on the benchmarks themselves here.

if the addtl separators is a bit complicated can do in another PR.

pls rebase on master as lots of PEP changes recently and do a ``git diff master | flake8 --diff`` check

ping if you want to merge now (else, pls open a new issue for the other sep chars if not)

**Comment by chris-b1** [comment]
@jreback - updated for that asv note.  I was most of the way there already, so also updated to handle 
 1. date separator can be any of `{'-', '.', '/', '\\', ' '}`, as long as consistent
 2. leading 0s can also be omitted from the time components (must have `:` sep)

Passes the current test suite (adjusted for a couple timezone changes), not sure if there are other edge cases to try?

a couple perf regressions to figure out too

```
    before     after       ratio
  [567bc5ce] [82c5d7b9]
+  154.12ms      4.24s     27.50  plotting.plot_timeseries_period.time_plot_timeseries_period
+    1.82ms      2.19s   1203.52  timeseries.datetimeindex_infer_dst.time_datetimeindex_infer_dst
-     2.10s     5.14ms      0.00  timeseries.timeseries_to_datetime_iso8601.time_timeseries_to_datetime_iso8601_nosep
-     3.13s   223.79ms      0.07  timeseries.timeseries_to_datetime_iso8601.time_timeseries_to_datetime_iso8601_tz_spaceformat
```

File: pandas/src/datetime/np_datetime_strings.c [hunk 2]
```diff
@@ -664,7 +722,7 @@ parse_iso_8601_datetime(char *str, int len,
     if (sublen >= 2 && isdigit(substr[0]) && isdigit(substr[1])) {
         out->min = 10 * (substr[0] - '0') + (substr[1] - '0');
 
-        if (out->hour < 0 || out->min >= 60) {
+        if (out->min < 0 || out->min >= 60) {
```
Comment (kawochen) [inline pandas/src/datetime/np_datetime_strings.c]: is `out->min < 0` ever `True`?

File: pandas/tseries/tools.py [hunk 3]
```diff
@@ -195,16 +193,13 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
         - If 'ignore', then invalid parsing will return the input
     dayfirst : boolean, default False
         Specify a date parse order if `arg` is str or its list-likes.
-        If True, parses dates with the day first, eg 10/11/12 is parsed as
-        2012-11-10.
+        If True, parses dates with the day first, eg 10/11/12 is parsed as 2012-11-10.
```
Comment (jorisvandenbossche) [inline pandas/tseries/tools.py]: Some changes from the PEP8 cleanup that got reverted here I think

Comment (jorisvandenbossche) [inline pandas/tseries/tools.py]: also in the rest of this file

**Comment by chris-b1** [comment]
@jreback  - if you want to have a look, I've got everything working now.

**Comment by jreback** [comment]
can you add tests for differnt seps (and invalid ones)?

File: pandas/src/datetime/np_datetime_strings.c [hunk 4]
```diff
@@ -546,12 +558,19 @@ parse_iso_8601_datetime(char *str, int len,
...
+            if (*substr == valid_sep[i]) {
+                has_sep = 1;
+                sep = valid_sep[i];
+                ++substr;
+                --sublen;
+                break;
+            }
+        }
+        if (i == valid_sep_len) {
+            goto parse_error;
```
Comment (jreback) [inline pandas/src/datetime/np_datetime_strings.c]: does this validate all seps are the same?

Comment (chris-b1) [inline pandas/src/datetime/np_datetime_strings.c]: This first check (sep between year and month) validates its in the list - the second check (between month/day) is required to match `sep`

File: pandas/tslib.pyx [hunk 5]
```diff
@@ -1388,6 +1388,26 @@ cpdef convert_str_to_tsobject(object ts, object tz, object unit,
 
     return convert_to_tsobject(ts, tz, unit)
 
+def _parse_iso8601(object ts):
+    '''
+    Parse string into Timestamp using iso8601 parser. Used ONLY for
```
Comment (jreback) [inline pandas/tslib.pyx]: why is this necessary? can't you just ``Timestamp(...)``?

**Comment by jreback** [comment]
did you figure out the perf regressions?

File: pandas/tslib.pyx (hunk 5 above)
Comment (chris-b1) [inline pandas/tslib.pyx]: It's not completely necessary, but only way to test certain invalid raise in the iso parsing, since `Timestamp(...)` falls back to to dateutil.  For example YYYYMM is not valid iso, but dateutil will parse if it matches YYMMDD -

```
In [1]: pd.Timestamp('200101')                
Out[1]: Timestamp('2020-01-01 00:00:00')      
```

Comment (jreback) [inline pandas/tslib.pyx]: can you move it to ``testing.pyx``? and maybe call it ``_test_parse_iso8601`` or something to make it completely obvious that it should not be used except in testing.

Comment (jreback) [inline pandas/tslib.pyx]: nix moving it, just rename and make the warning bigger in the doc-string

**Comment by chris-b1** [comment]
Well, I seemed to fix it, but didn't figure it out.  

`tz_localize` was very running very slow, but only a fresh install of pandas - wasn't slow if run a second time.  Also seemed to be fine if stepping through with `pdb`.  Undoing this [change](https://github.com/chris-b1/pandas/commit/5d565297cc653c80d0c3eaece864d457135c6876#diff-05de2950b2c86c9828531957e02d1b87L2021) fixed it - very unclear to me how that caused it though.

**Comment by jreback** [comment]
hmm, I think this interferes with the ``infer_datetime_format`` ? haven't really stepped thru the code that recently.

**Comment by chris-b1** [comment]
@jreback - updated.  Perf issue turned out to be unrelated - just some kind of caching that `pytz` does (also shows up on a fresh install of 0.17.1).

```
    before     after       ratio
  [5b5b2fe8] [5be82023]
   925.61us   929.00us      1.00  frame_methods.frame_assign_timeseries_index.time_frame_assign_timeseries_index
   145.86ms   145.63ms      1.00  plotting.plot_timeseries_period.time_plot_timeseries_period
     3.18ms     3.11ms      0.98  timeseries.dataframe_resample_max_numpy.time_dataframe_resample_max_numpy
     3.21ms     3.10ms      0.97  timeseries.dataframe_resample_max_string.time_dataframe_resample_max_string
     2.79ms     2.73ms      0.98  timeseries.dataframe_resample_mean_numpy.time_dataframe_resample_mean_numpy
     2.76ms     2.86ms      1.03  timeseries.dataframe_resample_mean_string.time_dataframe_resample_mean_string
     3.40ms     3.25ms      0.96  timeseries.dataframe_resample_min_numpy.time_dataframe_resample_min_numpy
     3.40ms     3.25ms      0.95  timeseries.dataframe_resample_min_string.time_dataframe_resample_min_string
   667.96us   661.20us      0.99  timeseries.datetimeindex_add_offset.time_datetimeindex_add_offset
     1.19ms     1.16ms      0.98  timeseries.datetimeindex_converter.time_datetimeindex_converter
     1.89ms     1.86ms      0.98  timeseries.datetimeindex_infer_dst.time_datetimeindex_infer_dst
     2.21ms     2.40ms      1.08  timeseries.datetimeindex_normalize.time_datetimeindex_normalize
   131.10us   127.90us      0.98  timeseries.datetimeindex_unique.time_datetimeindex_unique
   401.88us   395.18us      0.98  timeseries.dti_reset_index.time_dti_reset_index
   445.96us   462.25us      1.04  timeseries.dti_reset_index_tz.time_dti_reset_index_tz
    63.58ms    67.03ms      1.05  timeseries.period_setitem.time_period_setitem
     1.04ms     1.07ms      1.03  timeseries.timeseries_1min_5min_mean.time_timeseries_1min_5min_mean
     1.24ms     1.23ms      0.99  timeseries.timeseries_1min_5min_ohlc.time_timeseries_1min_5min_ohlc
    13.13ms    13.95ms      1.06  timeseries.timeseries_add_irregular.time_timeseries_add_irregular
     3.07ms     3.04ms      0.99  timeseries.timeseries_asof.time_timeseries_asof
     2.92ms     2.93ms      1.00  timeseries.timeseries_asof_nan.time_timeseries_asof_nan
    42.46us    43.79us      1.03  timeseries.timeseries_asof_single.time_timeseries_asof_single
    20.10us    23.67us      1.18  timeseries.timeseries_custom_bday_apply.time_timeseries_custom_bday_apply
    32.52us    33.25us      1.02  timeseries.timeseries_custom_bday_apply_dt64.time_timeseries_custom_bday_apply_dt64
    36.82us    41.39us      1.12  timeseries.timeseries_custom_bday_cal_decr.time_timeseries_custom_bday_cal_decr
    29.51us    30.93us      1.05  timeseries.timeseries_custom_bday_cal_incr.time_timeseries_custom_bday_cal_incr
    30.64us    31.04us      1.01  timeseries.timeseries_custom_bday_cal_incr_n.time_timeseries_custom_bday_cal_incr_n
    36.54us    35.03us      0.96  timeseries.timeseries_custom_bday_cal_incr_neg_n.time_timeseries_custom_bday_cal_incr_neg_n
    35.01us    34.81us      0.99  timeseries.timeseries_custom_bday_decr.time_timeseries_custom_bday_decr
    22.65us    24.13us      1.07  timeseries.timeseries_custom_bday_incr.time_timeseries_custom_bday_incr
   275.66us   262.08us      0.95  timeseries.timeseries_custom_bmonthbegin_decr_n.time_timeseries_custom_bmonthbegin_decr_n
   245.19us   255.52us      1.04  timeseries.timeseries_custom_bmonthbegin_incr_n.time_timeseries_custom_bmonthbegin_incr_n
   279.26us   292.99us      1.05  timeseries.timeseries_custom_bmonthend_decr_n.time_timeseries_custom_bmonthend_decr_n
   196.76us   202.95us      1.03  timeseries.timeseries_custom_bmonthend_incr.time_timeseries_custom_bmonthend_incr
   244.31us   237.53us      0.97  timeseries.timeseries_custom_bmonthend_incr_n.time_timeseries_custom_bmonthend_incr_n
     5.78ms     5.70ms      0.99  timeseries.timeseries_datetimeindex_offset_delta.time_timeseries_datetimeindex_offset_delta
    18.93ms    19.09ms      1.01  timeseries.timeseries_datetimeindex_offset_fast.time_timeseries_datetimeindex_offset_fast
    52.19ms    50.24ms      0.96  timeseries.timeseries_datetimeindex_offset_slow.time_timeseries_datetimeindex_offset_slow
    53.03us    59.37us      1.12  timeseries.timeseries_day_apply.time_timeseries_day_apply
    57.71us    58.96us      1.02  timeseries.timeseries_day_incr.time_timeseries_day_incr
    12.00ms    12.26ms      1.02  timeseries.timeseries_infer_freq.time_timeseries_infer_freq
     4.06ms     4.23ms      1.04  timeseries.timeseries_is_month_start.time_timeseries_is_month_start
   681.27ms   691.78ms      1.02  timeseries.timeseries_iter_datetimeindex.time_timeseries_iter_datetimeindex
    12.82ms    12.77ms      1.00  timeseries.timeseries_iter_datetimeindex_preexit.time_timeseries_iter_datetimeindex_preexit
      3.71s      3.65s      0.98  timeseries.timeseries_iter_periodindex.time_timeseries_iter_periodindex
    39.32ms    37.26ms      0.95  timeseries.timeseries_iter_periodindex_preexit.time_timeseries_iter_periodindex_preexit
    65.22us    65.56us      1.01  timeseries.timeseries_large_lookup_value.time_timeseries_large_lookup_value
    17.02ms    16.28ms      0.96  timeseries.timeseries_period_downsample_mean.time_timeseries_period_downsample_mean
     2.52ms     2.99ms      1.18  timeseries.timeseries_resample_datetime64.time_timeseries_resample_datetime64
     7.88ms     7.79ms      0.99  timeseries.timeseries_series_offset_delta.time_timeseries_series_offset_delta
    21.77ms    21.16ms      0.97  timeseries.timeseries_series_offset_fast.time_timeseries_series_offset_fast
    54.98ms    53.63ms      0.98  timeseries.timeseries_series_offset_slow.time_timeseries_series_offset_slow
    48.59us    47.74us      0.98  timeseries.timeseries_slice_minutely.time_timeseries_slice_minutely
     9.76ms     9.74ms      1.00  timeseries.timeseries_sort_index.time_timeseries_sort_index
     8.43ms     8.26ms      0.98  timeseries.timeseries_timestamp_downsample_mean.time_timeseries_timestamp_downsample_mean
    17.30us    16.55us      0.96  timeseries.timeseries_timestamp_tzinfo_cons.time_timeseries_timestamp_tzinfo_cons
    11.15ms    11.18ms      1.00  timeseries.timeseries_to_datetime_YYYYMMDD.time_timeseries_to_datetime_YYYYMMDD
     4.54ms     4.81ms      1.06  timeseries.timeseries_to_datetime_iso8601.time_timeseries_to_datetime_iso8601
     6.95ms     4.70ms      0.68  timeseries.timeseries_to_datetime_iso8601.time_timeseries_to_datetime_iso8601_format
-  114.53ms     4.67ms      0.04  timeseries.timeseries_to_datetime_iso8601.time_timeseries_to_datetime_iso8601_format_no_sep
-     1.98s     4.65ms      0.00  timeseries.timeseries_to_datetime_iso8601.time_timeseries_to_datetime_iso8601_nosep
-     2.91s   210.18ms      0.07  timeseries.timeseries_to_datetime_iso8601.time_timeseries_to_datetime_iso8601_tz_spaceformat
```

File: pandas/src/datetime/np_datetime_strings.c [hunk 6]
```diff
@@ -695,7 +753,7 @@ parse_iso_8601_datetime(char *str, int len,
     if (sublen >= 2 && isdigit(substr[0]) && isdigit(substr[1])) {
         out->sec = 10 * (substr[0] - '0') + (substr[1] - '0');
 
-        if (out->sec < 0 || out->sec >= 60) {
```
Comment (jreback) [inline pandas/src/datetime/np_datetime_strings.c]: are these negative checks still needed?

File: pandas/src/datetime/np_datetime_strings.c [hunk 7]
```diff
@@ -396,6 +394,16 @@ parse_iso_8601_datetime(char *str, int len,
     char *substr, sublen;
     PANDAS_DATETIMEUNIT bestunit;
 
+    /* if date components in are seperated by one of valid separators
```
Comment (jreback) [inline pandas/src/datetime/np_datetime_strings.c]: separated (a couple of places I think)

File: pandas/tseries/tests/test_tslib.py [hunk 8]
```diff
@@ -688,6 +688,28 @@ def test_parsers_timezone_minute_offsets_roundtrip(self):
             converted_time = dt_time.tz_localize('UTC').tz_convert(tz)
             self.assertEqual(dt_string_repr, repr(converted_time))
 
+    def test_parsers_iso8601(self):
+        # GH 12060
```
Comment (jreback) [inline pandas/tseries/tests/test_tslib.py]: give a comment about what we are testing & doing here (and that all seps must be the same)

File: pandas/tseries/tools.py [hunk 9]
```diff
@@ -461,6 +458,19 @@ def calc_with_mask(carg, mask):
 
     return None
 
+def _format_is_iso(f):
+    """
+    Does format match the iso8601 set that can be handled by the C parser?
```
Comment (jreback) [inline pandas/tseries/tools.py]: add here that we require same seps for date fields

**Comment by jreback** [comment]
some minor comments. ping when pushed / green.

File: pandas/src/datetime/np_datetime_strings.c (hunk 6 above)
Comment (chris-b1) [inline pandas/src/datetime/np_datetime_strings.c]: @kawochen pointed this out - unless I'm misunderstanding something they will always be false.

**Comment by chris-b1** [comment]
@jreback  - updated for your comments

**Comment by jreback** [comment]
@chris-b1 thanks!

your PR's are always great!
